vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -1507,7 +1510,24 @@ private void setRequestedRange(int start, int length, String filter) {
 
     @ClientCallable
     private void resetDataCommunicator() {
-        dataCommunicator.reset();
+        /*
+         * The client filter from combo box will be used in the data
+         * communicator only within 'setRequestedRange' calls to data provider,
+         * and then will be erased to not affect the data view item count
+         * handling methods. Thus, if the current client filter is not empty,
+         * then we need to re-set it in the data communicator.
+         */
+        if (lastFilter == null || lastFilter.isEmpty()) {
+            dataCommunicator.reset();
+        } else {
+            String filter = lastFilter;
+            lastFilter = null;
+            /*
+             * Call to 'DataCommunicator::reset' is done inside the filter
+             * slot callback, so we don't need to explicitly call it.
+             */
+            filterSlot.accept(filter);
```

Comment:
Actually now I realized that there is no `dataCommunicator::reset` call in the filter slot callback (lines 867-870), instead it is just resetting the existing filter, so the comment above is not entirely true and thus a bit confusing.

Reply:
Yeah, you're right, actually `dataCommunicator::reset` is on one level lower, in the data communicator's filter consumer, returned by `setDataProvider` method: https://github.com/vaadin/flow/blob/b250bce9dbc5da986ca2281345c4c506eca1f5b2/flow-data/src/main/java/com/vaadin/flow/data/provider/DataCommunicator.java#L342

Reply:
Then the comment above should reflect that ?

Reply:
Yeah, I forgot about it :) My last commit fixes that.